### DISCUSSION TITLE
script outside of the html

### DISCUSSION THREAD

**Discussion by valentinedwv** [mafredri/cdp#137] [Q&A] [answered]
We have a page where the script is outside of the HTML
```
<html>
<head>
</head>
<body></body>
</html>
<script></script>

```
Best shown on the [validator.schema.org](https://validator.schema.org/#url=https%3A%2F%2Fdev.rvdata.us%2Fsearch%2Ffileset%2F100137)

In chrome developers tools pane in the chrome browser, the script is move into the head.
```
<html>
<head>
<script></script>
</head>
<body></body>
</html>
```

Is it possible to do this in the cdp tooling?

**Comment by valentinedwv** [comment, marked as answer]
actually, it looks like  the javascript code should be rendering to the head, but when using go library, the  element is not found there.
```
    h.type = "application/ld+json",
        g = f.trustAsHtml(e("json")(g)),
        h.text = g,
        document.querySelector("head").appendChild(h)

```

Using LoadEventFired does let the script load into the head.

**Reply by mafredri** [reply]
Hi, are you using this Go library or just Chrome DevTools Protocol in general? If you are trying to achieve this with the Go library, could you give a code sample of what you tried to do that didn’t work?

If it’s not related to this Go library, I’m afraid I can’t provide any assistance.

**Reply by valentinedwv** [reply]
Yes, using the go library, code is [ here](https://github.com/gleanerio/gleaner/blob/c180764b588aae2e3b20c6e1cc8cb6774106ed7b/internal/summoner/acquire/headlessNG.go#L215)

Should we wait longer after the domContent.Recv() event? or is there a way to wait for the page rendered?

**Reply by mafredri** [reply]
Hey @valentinedwv, I'm in the middle of moving so my responses are pretty slow, sorry about that. 

Thanks for sharing the code, I don't really see a reason it shouldn't work, but instead of (or in addition to) using `DOMContentEventFired`, have you considered waiting until [`LoadEventFired`](https://pkg.go.dev/github.com/mafredri/cdp@v0.33.0/protocol/page#LoadEventFiredClient)? It sounds more suitable for your purpose: https://developer.mozilla.org/en-US/docs/Web/API/Window/load_event.

Another option may be to enable lifecycle events and check if they provide anything useful: https://pkg.go.dev/github.com/mafredri/cdp@v0.33.0/protocol/page#LifecycleEventClient

**Reply by valentinedwv** [reply]
Thanks. Yes, looks like LoadEventFired works.
